imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -271,3 +291,88 @@ def pset_geometry(pset_et: float, sensor_str: str) -> dict[str, xr.DataArray]:
         np.newaxis, :
     ]
     return geometry_vars
+
+
+@pd.api.extensions.register_dataframe_accessor("cal_prod_config")
+class CalibrationProductConfig:
+    """
+    Register custom accessor for calibration product configuration DataFrames.
+
+    Parameters
+    ----------
+    pandas_obj : pandas.DataFrame
+        Object to run validation and use accessor functions on.
+    """
+
+    required_columns = (
+        "cal_prod_num",
+        "esa_energy_step",
+        "coincidence_type_list",
+        "tof_ab_low",
+        "tof_ab_high",
+        "tof_ac1_low",
+        "tof_ac1_high",
+        "tof_bc1_low",
+        "tof_bc1_high",
+        "tof_c1c2_low",
+        "tof_c1c2_high",
+    )
+
+    def __init__(self, pandas_obj: pd.DataFrame) -> None:
+        self._validate(pandas_obj)
+        self._obj = pandas_obj
+
+    def _validate(self, df: pd.DataFrame) -> None:
+        """
+        Validate the current configuration.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            Object to validate.
+
+        Raises
+        ------
+        AttributeError : If the dataframe does not pass validation.
+        """
+        # Verify that the Dataframe has all the required columns
+        for col in self.required_columns:
+            if col not in df.columns:
+                raise AttributeError(f"Required column {col} not present in dataframe.")
+        # TODO: Verify that the same ESA energy steps exist in all unique calibration
+        #   product numbers
+
+    @classmethod
+    def read_csv(cls, path: Path) -> pd.DataFrame:
+        """
+        Read configuration CSV file into a pandas.DataFrame.
+
+        Parameters
+        ----------
+        path : Path
+            Location of the Calibration Product configuration CSV file.
+
+        Returns
+        -------
+        dataframe : pandas.DataFrame
+            Validated calibration product configuration data frame.
+        """
+        return pd.read_csv(
+            path,
+            index_col=False,
```

Comment:
Would it be helpful to make this a multi-index, so you can lookup based on `(cal_prod, esa_step)` later on? If you read it in as a sequence a multi-index will be created.
```suggestion
            index_col=[0, 1],
```

Reply:
Yes, great suggestion. Thanks!